### ISSUE TITLE
Switching from Apigee Edge to Apigee X causes mismatched Entity/Field Definitions

### ISSUE THREAD

**Issue by psweeney-achieve** [apigee/apigee-edge-drupal#1013]
## Description
Whenever we change our backend from Apigee Edge to Apigee X via `/admin/config/apigee-edge/settings`, we receive these errors on the Status report page:

<img width="881" alt="Screenshot 2023-12-20 at 9 49 52 AM" src="https://github.com/apigee/apigee-edge-drupal/assets/94869238/e83e9ef3-175c-44a2-b24a-189ff1ddfeca">

## Apigee Info
Apigee X & Edge

## Steps to Reproduce
1. Go to '/admin/config/apigee-edge/settings'
2. Enter Apigee X credentials and save
3. Navigate to `/admin/reports/status`
4. View errors

## Actual Behavior
Mismatched entity / field definition errors is displayed on the Status Report page and Entity Update conrib module must be used to manually update field definitions.

## Expected Behavior
After enabling Apigee X, the base entity and field definitions should be synced properly.

## Version Info
Drupal D10/9 and Apigee Edge 2.1.3 (D9) or Apigee Edge 3.0.4 (D10)

**Comment by shishir-intelli** [comment]
@psweeney-achieve 
Yes, this is a known issue when switching the organization from `Apigee Edge` to `Apigee X` and vice-versa.
Thank you for bringing up this concern. we're working on fixing this issue.
